compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -395,6 +395,9 @@ public BSymbol resolvePrefixSymbol(SymbolEnv env, Name pkgAlias, Name compUnit)
             }
 
             if ((entry.symbol.tag & SymTag.IMPORT) == SymTag.IMPORT &&
```

Comment:
Shall we refactor this into something like

```java
            if ((entry.symbol.tag & SymTag.IMPORT) == SymTag.IMPORT) {
                Name importCompUnit = ((BPackageSymbol) entry.symbol).compUnit;
                if (importCompUnit == null) {
                    return entry.symbol;
                } else if (importCompUnit.equals(compUnit)) {
                    ((BPackageSymbol) entry.symbol).isUsed = true;
                    return entry.symbol;
                }
            }
```

Reply:
Let's also add a comment to say why the compUnit can be null here?

Reply:
Done